sydent/db/sqlitedb.py

```diff
@@ -186,6 +186,17 @@ def _upgradeSchema(self):
             logger.info("v3 -> v4 schema migration complete")
             self._setSchemaVersion(4)
 
+        if curVer < 5:
+            # Remove all associations from the database that have a NULL MXID
+            # We set the MXID to NULL when we removed the binding but we've switched it to
+            # removing the whole row to minimize personal information
+            # https://github.com/matrix-org/sydent/issues/192
+            cur = self.db.cursor()
+            cur.execute("DELETE FROM local_threepid_associations WHERE mxid = NULL")
```

Comment:
```suggestion
            cur.execute("DELETE FROM local_threepid_associations WHERE mxid IS NULL")
```

`X = null` always results in `null`, event when `X` is `null`. In general `null` is not considered equal to `null`, except for some purposes like `DISTINCT` (e.g. unique indices can have multiple null values)

Reply:
oh, very good to know